Source/Core/VideoCommon/RenderState.h

```diff
@@ -101,12 +103,12 @@ union SamplerState
   BitField<2, 1, Filter> mipmap_filter;
   BitField<3, 2, AddressMode> wrap_u;
   BitField<5, 2, AddressMode> wrap_v;
-  BitField<7, 8, u32> min_lod;    // multiplied by 16
-  BitField<15, 8, u32> max_lod;   // multiplied by 16
-  BitField<23, 8, s32> lod_bias;  // multiplied by 32
-  BitField<31, 1, u32> anisotropic_filtering;
+  BitField<7, 16, s32> lod_bias;              // multiplied by 256
+  BitField<23, 8, u32> min_lod;               // multiplied by 16
+  BitField<31, 8, u32, StorageType> max_lod;  // multiplied by 16
```

Comment:
This is likely to invoke undefined behaviour when these members are accessed. `max_lod` and `anisotropic_filtering` are not layout-compatible with the other members of this `union`.

Reply:
As far as I know it's legal to have members with different sizes in an union?

Reply:
It is, only if the different sized types are used on their own and don't read out of other differently sized members when they're not active (in this case the struct is used as a 'dumb' bit extractor). This would break the rule about common initial sequences.

small re-cap:

given structs (or classes):

```cpp
struct A
{
  uint16_t a;
  uint32_t b;
};

struct B
{
  uint16_t a;
  uint32_t b;
};

struct C
{
  uint64_t a;
  uint16_t b;
};
```

and a union

```cpp
union U
{
  A x;
  B y;
  C z;
};
```

If `x` or `y` is the active union member, any of `a` or `b` from either `x` or `y` can be inspected, since they have a common initial sequence. However, none of `z`'s members may be accessed, as the struct `C` doesn't have a compatible initial sequence.

A `BitField` at it's core is really just a:

```cpp
template <typename T>
struct BitField
{
  T data;
};
```

so as long as all members are the same type, any access to any members in the union can be made. However, your change makes the internal data type a `StorageType` which is an alias for a `u64`, which isn't layout compatible with a `u32` or `s32`.

If `anisotropic_filtering` or `max_lod` were made the active member of the union, no other `BitField` instance other than those two are ever allowed to be accessed, because they too are not layout compatible with any of the other `BitField` instances that are either `u32` or `s32` internally.

Reply:
Oh ok. I can just make everything `StorageType`. There is only one issue there, `lod_bias` breaks if you make the StorageType be a signed type.
That seems to be a preexisting issue as well, because I don't see why the storage type should affect the signedness of the represented type!

Would it be ok if everything but `lod_bias` was `StorageType`, while `lod_bias` is `i64`? Otherwise, I can try to fix `BitField` and remove the signedness "leak".